### ISSUE TITLE
Random Server Crash 1.21

### ISSUE THREAD

**Issue by fantahund** [PaperSpigot/Paper#10908]
### Stack trace

```
[12:00:12] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.ReportedException: Exception ticking world
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1744) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:471) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1538) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1246) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:329) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.IllegalStateException
	at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.ChunkHolderManager.processUnloads(ChunkHolderManager.java:1100) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.level.ChunkMap.processUnloads(ChunkMap.java:472) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.level.ChunkMap.tick(ChunkMap.java:461) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.level.ServerChunkCache.tick(ServerChunkCache.java:487) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:688) ~[paper-1.21.jar:1.21-1285-b4b8208]
	at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1733) ~[paper-1.21.jar:1.21-1285-b4b8208]
	... 5 more
[12:00:12] [Server thread/ERROR]: This crash report has been saved to: /home/minecraft/Server/Challenge-6/crash-reports/crash-2024-06-18_12.00.12-server.txt
[12:00:12] [Server thread/INFO]: Stopping server
```

### Plugin and Datapack List

/

### Actions to reproduce (if known)

It's probably not directly due to that, the server crashed after the 2 players who were online went into the Nether via a portal.

### Paper version

paper-1.21.jar:1.21-1285-b4b8208

**Comment by Quackster** [comment]
I recommend attaching the file it created: ```/home/minecraft/Server/Challenge-6/crash-reports/crash-2024-06-18_12.00.12-server.txt```

**Comment by fantahund** [comment]
https://cpaste.de/ezeharebob.yaml

**Comment by MrPowerGamerBR** [comment]
Can confirm the same issue, but just like @fantahund I'm not really sure what causes this crash

```
[01:47:52] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.RuntimeException: java.util.concurrent.ExecutionException: net.minecraft.ReportedException: Exception ticking world
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1792) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:488) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1543) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1251) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:334) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.util.concurrent.ExecutionException: net.minecraft.ReportedException: Exception ticking world
        at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
        at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1789) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        ... 5 more
Caused by: net.minecraft.ReportedException: Exception ticking world
        at net.minecraft.server.MinecraftServer.lambda$tickChildren$20(MinecraftServer.java:1779) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        ... 1 more
Caused by: java.lang.IllegalStateException
        at ca.spottedleaf.moonrise.patches.chunk_system.scheduling.ChunkHolderManager.processUnloads(ChunkHolderManager.java:1100) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.level.ChunkMap.processUnloads(ChunkMap.java:477) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.level.ChunkMap.tick(ChunkMap.java:466) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.level.ServerChunkCache.tick(ServerChunkCache.java:511) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:696) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at net.minecraft.server.MinecraftServer.lambda$tickChildren$20(MinecraftServer.java:1747) ~[sparklypaper-1.21.jar:1.21-DEV-1b63d66]
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[?:?]
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        ... 1 more
[01:47:52] [Server thread/ERROR]: This crash report has been saved to: /home/survival/crash-reports/crash-2024-06-19_01.47.52-server.txt
[01:47:52] [Server thread/INFO]: Stopping server
```

**Comment by thedig123** [comment]
using paperMC experimental (1.21) spotted.moonrise.patches seems to causing issues for my friend and I our world reverted back to when I roughly logged into the server judging by the time difference (when it crashed it went back to before I placed a chest with my inventory, cut down some trees and made a 2 tunnel wide mineshaft with a friend down to y=6, when it undid that I lost my chest with my inventory)

the crash happened when i triggered a skulk sensor and my friend at the time was taking taking damage, although that might just be random.

my player data was fine aka i had all the diamonds I mined and redstone and such from that mining session in the cave

[crash-2024-06-18_23.14.17-server.txt](https://github.com/user-attachments/files/15906431/crash-2024-06-18_23.14.17-server.txt)

**Comment by MrPowerGamerBR** [comment]
> using paperMC experimental (1.21) spotted.moonrise.patches seems to causing issues for my friend and I our world reverted back to when I roughly logged into the server judging by the time difference (when it crashed it went back to before I placed a chest with my inventory, cut down some trees and made a 2 tunnel wide mineshaft with a friend down to y=6, when it undid that I lost my chest with my inventory)
> 
> the crash happened when i triggered a skulk sensor and my friend at the time was taking taking damage, although that might just be random.
> 
> my player data was fine aka i had all the diamonds I mined and redstone and such from that mining session in the cave
> 
> [crash-2024-06-18_23.14.17-server.txt](https://github.com/user-attachments/files/15906431/crash-2024-06-18_23.14.17-server.txt)

Update Paper, there were multiple chunk fixes in newer builds

**Comment by Machine-Maker** [comment]
fixed in 19105a9019d7e67fac176dd9b9709cd25c213d31
